source/Octopus.Client/Repositories/Async/BuiltInPackageRepositoryRepository.cs

```diff
@@ -62,18 +71,31 @@ public async Task<PackageFromBuiltInFeedResource> PushPackage(string fileName, S
                 Logger.Info("Pushing the complete package to the server, as delta compression was explicitly disabled");
             }
 
+            var link = await repository.Link("PackageUpload").ConfigureAwait(false);
+            object pathParameters;
+
+            // if the link doesn't contain overwritemode then we're connected to an older server, which uses the `replace` parameter  
```

Comment:
This comment makes me think you initially wanted to do `if (!link.Contains)` here...?